h2o-py/tests/testdir_algos/automl/pyunit_automl_model_selection.py

```diff
@@ -0,0 +1,259 @@
+
+
+def test_include_exclude_algos():
+    print("include_algos and exclude_algos parameters are mutually exclusive")
+    try:
+        H2OAutoML(project_name="py_include_exclude_algos",
+                  exclude_algos=['DRF', 'XGBoost'],
+                  include_algos=['GBM'],
+                  max_models=max_models,
+                  seed=1)
+        assert False, "Should have thrown AssertionError"
+    except AssertionError as e:
+        assert "Use either include_algos or exclude_algos" in str(e)
+
+
+def test_bad_training_plan_using_full_syntax():
+    try:
+        H2OAutoML(training_plan=[
+            dict(steps=['def_1'])
+        ])
+    except AssertionError as e:
+        assert "each definition must have a 'name' key" in str(e)
+
+    try:
+        H2OAutoML(training_plan=[
+            dict(name="GBM", steps=['def_1'], alias='defaults')
+        ])
+    except AssertionError as e:
+        assert "each definition must have only 1 or 2 keys" in str(e)
+
+    try:
+        H2OAutoML(training_plan=[
+            dict(name="GBM", alias='all_steps')
+        ])
+    except AssertionError as e:
+        assert "alias must be one of ['all', 'defaults', 'grids']" in str(e)
+
+    try:
+        H2OAutoML(training_plan=[
+            dict(name="GBM", dummy=['def_1'])
+        ])
+    except AssertionError as e:
+        assert "steps definitions support only the following keys: name, alias, steps" in str(e)
+
+    try:
+        H2OAutoML(training_plan=[
+            dict(name="GBM", steps=['def_1'])
+        ])
+    except AssertionError as e:
+        assert "each step must be a dict" in str(e)
+
+    try:
+        H2OAutoML(training_plan=[
+            dict(name="GBM", steps=[dict(foo='def_1')])
+        ])
+    except AssertionError as e:
+        assert "each step must have an 'id' key" in str(e)
+
+    try:
+        H2OAutoML(training_plan=[
+            dict(name="GBM", steps=[dict(id='def_1', weight=3/4)])
+        ])
+    except AssertionError as e:
+        assert "weight must be an integer" in str(e)
+
+
+def test_bad_training_plan_using_simplified_syntax():
+    try:
+        H2OAutoML(training_plan=[
+            ['GBM']
+        ])
+    except h2o.exceptions.H2OTypeError:
+        pass
+
+    try:
+        H2OAutoML(training_plan=[
+            ('GBM', 'defaults', ['def_1'])
+        ])
+    except AssertionError:
+        pass
+
+    try:
+        H2OAutoML(training_plan=[
+            ('GBM', 'dummy_alias')
+        ])
+    except h2o.exceptions.H2OTypeError:
+        pass
+
+    try:
+        H2OAutoML(training_plan=[
+            ('GBM', ('def_1', 'def_2'))
+        ])
+    except h2o.exceptions.H2OTypeError:
+        pass
+
+
+def test_training_plan_using_full_syntax():
+    ds = import_dataset()
+    aml = H2OAutoML(project_name="py_training_plan_full_syntax",
+                    max_models=2,
+                    training_plan=[
+                        dict(name='GLM', steps=[dict(id='def_1')]),
```

Comment:
@seb-h2o How would users know how many default models we have inside? only 'training_step' is a source of the knowledge probably.

Reply:
he can know by running it fully once and looking at `trained_steps`: we can also document this, but I would not suggest to document internals.
That's the main reason for aliases, to care less about internals.